components/admin_console/system_user_detail/team_list/team_row.jsx

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import {FormattedMessage} from 'react-intl';
+
+import * as Utils from 'utils/utils';
+
+import TeamImage from './team_image.jsx';
+
+export default class TeamRow extends React.Component {
+    static propTypes = {
+        team: PropTypes.object.isRequired,
+        onRowClick: PropTypes.func.isRequired,
+    };
+
+    handleRowClick = () => {
+        const {team, onRowClick} = this.props;
+        onRowClick(team.id);
+    }
+    renderTeamType = (team) => {
+        if (team.group_constrained) {
+            return (
+                <FormattedMessage
+                    id={'admin.systemUserDetail.teamList.teamType.groupSync'}
+                    defaultMessage={'Group Sync'}
+                />
+            );
+        }
+        if (team.allow_open_invite) {
+            return (
+                <FormattedMessage
+                    id={'admin.systemUserDetail.teamList.teamType.anyoneCanJoin'}
+                    defaultMessage={'Anyone can join'}
+                />
+            );
+        }
+        return (
+            <FormattedMessage
+                id={'admin.systemUserDetail.teamList.teamType.inviteOnly'}
+                defaultMessage={'Invite only'}
+            />
+        );
+    }
+    renderTeamRole = (team) => {
+        if (team.scheme_admin) {
+            return (
+                <FormattedMessage
+                    id={'admin.systemUserDetail.teamList.teamRole.admin'}
+                    defaultMessage={'Admin'}
+                />
+            );
+        }
+        return (
+            <FormattedMessage
+                id={'admin.systemUserDetail.teamList.teamRole.member'}
+                defaultMessage={'Member'}
+            />
+        );
+    }
+    render = () => {
+        const {team} = this.props;
+        const teamIconUrl = Utils.imageURLForTeam(team);
+        return (
+            <div
+                className={'group '}
+                onClick={this.handleRowClick}
+            >
+                <div className='group-row group-row-large'>
+                    <div className='group-name'>
+                        <div className='col-sm-auto'>
+                            <TeamImage
+                                small={true}
+                                teamIconUrl={teamIconUrl}
+                                displayName={team.display_name}
+                            />
+
+                        </div>
+                        <div className='col-sm-auto'>
+                            <b>{team.display_name}</b>
+                            {team.description && (
+                                <div
+                                    className='overflow--ellipsis text-nowrap'
+                                    style={{maxWidth: 500}}
```

Comment:
this needs to move to css

Reply:
I also noticed when looking at the original components that they aren't following the .scss import pattern that is being defined elsewhere. I believe @jespino has more information on that topic. In summary, there's an effort to import a component-specific .scss file (ie `.my_component.scss`) with a `.MyComponent` class defined.

Reply:
We are doing that effort mainly in the reusable components, but anywhere that we can use it, would be nice to use it (that will free a bit the already crowded main sass code, and leave that responsibility to each component).